include/onnxruntime/core/framework/data_types.h

```diff
@@ -76,6 +76,22 @@ using CreateFunc = void* (*)();
  *
  */
 class DataTypeImpl {
+ public:
+  enum GeneralType {
```

Comment:
these look like flag values - are they intended to support being combined? e.g., for an optional tensor or optional tensor sequence

any reason for skipping the value `1 << 0`?

Reply:
Good point. They don't need to be combined.